src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2074,6 +2081,27 @@ private void doSignalConnectTimeoutIfNecessary()
             }
         }
 
+        private int calculateSubscribeFlags(
+            long traceId,
+            long authorization,
+            int options)
+        {
+            int flags = 0;
+
+            final int retained = options & 0b0011_0000;
```

Comment:
Hoist `0b0011_0000` as constant called `RETAIN_HANDLING_MASK`.
Rename `retain` to `retainHandling`.